cmake/CMakeLists.txt

```diff
@@ -0,0 +1,123 @@
+cmake_minimum_required(VERSION 3.10)
+
+project(heatshrink C)
+
+# sources are actually one directory up
+set(srcdir ${PROJECT_SOURCE_DIR}/../)
+
+# When making a DLL with MSVC, symbols are not exported by default.
+# This can be changed with CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS but
+# it would really be better if an export header would be used instead.
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
+# Add separate libraries for lib instances with static and dynamic memory allocation 
+add_library(${PROJECT_NAME} ${srcdir}/heatshrink_decoder.c ${srcdir}/heatshrink_encoder.c)
```

Comment:
Why not have the suffix be explicit in both cases?

`_noalloc` vs. `dynalloc`?

Reply:
The naming in the makefile is _static and _dynamic. I find this unfortunate due to potential confusion with static or shared libraries (dynamic linking). If the project is named heatshrink, I would kind of expect that it provides a library with the same name. But yes, then it's hard to know which memory allocation is the library without the suffix using. Maybe I could add an alias to one of them as the default and export that as well.

Maybe this duplicity could also be avoided by using some kind of allocator abstraction. But that is out of the scope for this PR.

Reply:
Creating an alias sounds good.

Regarding abstracting the allocator: you'd need LTO for this to work properly and the two libs are a reasonable compromise. Also AFAIK there's some actual code differences in how the memory is used with the dynamic allocation, thus that's not something you could easily abstract away without compromising on the memory footprint.

Reply:
Actually while I'm thinking about this library naming, I guess even if the naming is confusing to me, it might still be better to replicate the naming from the original makefile. Because now, when you build with CMake you get differently named libraries installed. It should not matter which build system was used to install the libraries, the result should be the same.